### ISSUE TITLE
dnf_package bug

### ISSUE THREAD

**Issue by jaymzh** [chef/chef#8042]
## Description

It seems that the way Chef is using the DNF internal libraries doesn't get the same data as the CLI.

```
# dnf list avail testpackage1
Installed Packages
testpackage1.x86_64               20181128-133340                @fb-runtime
Available Packages
testpackage1.x86_64               20181128-133340                @fb-runtime
testpackage1.x86_64               20181202-044213                fb-runtime 
testpackage1.x86_64               20181202-063154                fb-runtime 
testpackage1.x86_64               20181203-063312                fb-runtime 
testpackage1.x86_64               20181203-100514                fb-runtime 
testpackage1.x86_64               20181204-063215                fb-runtime
```

Note the versions there. Now if I use the API chef does:

```
# /usr/libexec/platform-python
Python 3.6.6 (default, Oct 16 2018, 01:53:53) 
[GCC 8.2.1 20180905 (Red Hat 8.2.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dnf
>>> s=dnf.subject.Subject('testpackage1')
>>> base = dnf.Base()
>>> base.read_all_repos()
>>> base.fill_sack()
<dnf.sack.Sack object at 0x7fbac8862728>
>>> q=s.get_best_query(base.sack, with_provides=True)
>>> qq=q.available()
>>> x = qq.run()
>>> for i in x:
...   print(i)
... 
testpackage1-20181127-063148.x86_64
testpackage1-20181128-063204.x86_64
testpackage1-20181128-133340.x86_64
testpackage1-20181128-133433.x86_64
testpackage1-20181129-063158.x86_64
```

That means that Chef will bail out with "version not available" in the event we request a version that is not in the latter but is in the former.

Note the difference in versions there.

## Chef Version
Chef 13.10.0 - but we've repro'd on 14 as well.

## Platform Version
RHEL8 beta

**Comment by jaymzh** [comment]
It's worth noting that this may in fact be a RHEL8 or DNF bug... but my best guess is we're using the API wrong.

**Comment by davide125** [comment]
To clarify, we reproed on the RHEL 8 Beta, not on CentOS

**Comment by lamont-granquist** [comment]
I suspect this got fixed by #8754

New issue would need to get opened if I'm wrong.
